octodns/provider/base.py

```diff
@@ -60,7 +72,8 @@ def plan(self, desired):
 
         # allow the provider to filter out false positives
         before = len(changes)
-        changes = [c for c in changes if self._include_change(c)]
+        changes = [c for c in changes if self._include_change(c) and
+                   self._check_root_ns(c)]
```

Comment:
I think I'd prefer to add `_check_root_ns` logic to `_include_change` since that's the sort of thing it is intended to check, just didn't have anything to do in the base provider until now. Looks like the two existing places where `_include_change` is overridden don't currently call up to `super`, but they actually should, `return super(XXX, self)._include_change(change)` rather than `return True`.